File: ios/Validation/EditorRequestValidator.swift
import Foundation

enum EditorRequestValidator {
  static func errorCode(for request: EditorRequest) -> String? {
    if request.uri.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty { return "E_INVALID_URI" }
    if request.mediaType != "photo" && request.mediaType != "video" { return "E_UNSUPPORTED_MEDIA_TYPE" }
    return nil
  }
}

File: android/src/main/java/com/photovideoeditor/validation/EditorRequestValidator.kt
package com.photovideoeditor.validation

import com.photovideoeditor.model.EditorRequest

internal object EditorRequestValidator {
  fun validate(request: EditorRequest): String? = when {
    request.uri.isBlank() -> "E_INVALID_URI"
    request.mediaType != "photo" && request.mediaType != "video" -> "E_UNSUPPORTED_MEDIA_TYPE"
    else -> null
  }
}
